src/main/groovy/com/lesfurets/jenkins/unit/InterceptingGCL.groovy

```diff
@@ -18,11 +18,20 @@ class InterceptingGCL extends GroovyClassLoader {
             Map.Entry<MethodSignature, Closure> matchingMethod = helper.allowedMethodCallbacks.find { k, v -> k == signature }
             if (matchingMethod) {
                 // a matching method was registered, replace script method execution call with the registered closure (mock)
-                metaClazz."$scriptMethod.name" = matchingMethod.value ?: {}
+                metaClazz."$scriptMethod.name" = matchingMethod.value ?: defaultClosure(matchingMethod.key.args)
             }
         }
     }
 
+    static Closure defaultClosure(Class[] args) {
+        String argumentsString = args.inject("") { acc, value ->
+            acc + value.name + " " + "a"*(acc.count(',') + 1) + ","}
```

Comment:
IMHO this is more readable:

```suggestion
        String argumentsString = args.inject("") { acc, value ->
            return "${acc}${value.name} a${*(acc.count(',') + 1)},"
        }
```

Reply:
This change did not work out of the box, instead I did something similar. I hope I got the idea behind your suggestion.